2024/opmodes/other/OpenCVPipelineBlue.java

```diff
@@ -0,0 +1,141 @@
+package opmodes.other;
+
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+import com.qualcomm.robotcore.util.RobotLog;
+
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.Point;
+import org.opencv.core.Rect;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.imgproc.Moments;
+import org.openftc.easyopencv.OpenCvCamera;
+import org.openftc.easyopencv.OpenCvPipeline;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import team25core.DeadReckonTask;
+import team25core.Robot;
+import team25core.RobotEvent;
+
+@TeleOp(name = "OpenCVBlue")
+
+public class OpenCVPipelineBlue extends Robot {
+
+
+    double cX = 0;
+    double cY = 0;
+    double width = 0;
+
+    private OpenCvCamera controlHubCam;  // Use OpenCvCamera class from FTC SDK
+    private static final int CAMERA_WIDTH = 640; // width  of wanted camera resolution
+    private static final int CAMERA_HEIGHT = 360; // height of wanted camera resolution
+
+    // Calculate the distance using the formula
+    public static final double objectWidthInRealWorldUnits = 3.75;  // Replace with the actual width of the object in real-world units
+    public static final double focalLength = 728; // Replace with the focal length of the camera in pixels
+
+    public static final String detectionlocation = "";
+
+    @Override
+    public void handleEvent(RobotEvent e)
+    {
+    }
+    public Mat processFrame(Mat input) {
+        // Preprocess the frame to detect blue regions
+        Mat blueMask = preprocessFrame(input);
+
+        // Find contours of the detected red regions
+        List<MatOfPoint> contours = new ArrayList<>();
+        Mat hierarchy = new Mat();
```

Comment:
This is a potential crash scenario during a match.

See this warning...

leakMsg = String.format("OpenCV pipeline leaking memory @ approx. %dMB/sec; %d%% RAM currently free. DO NOT create new Mats or re-assign Mat variables inside processFrame()!", (int)leakRate, (int)availMemPercent);

From:  https://github.com/OpenFTC/EasyOpenCV/blob/8f879f6ad72586fe4177779f6d48b5f01d57f590/easyopencv/src/main/java/org/openftc/easyopencv/OpenCvPipeline.java#L149